tests/AutoReview/ProjectCodeTest.php

```diff
@@ -299,6 +315,51 @@ function ($item) {
         );
     }
 
+    public function provideTestClassNamesAndDataProviderMethodNames()
+    {
+        $data = array();
+
+        $classNames = $this->getTestClasses();
+
+        foreach ($classNames as $className) {
+            $reflection = new \ReflectionClass($className);
+
+            if (!$reflection->isSubclassOf('PHPUnit\Framework\TestCase')) {
+                continue;
+            }
+
+            $dataProviderMethodNames = array_unique(array_reduce(
+                $reflection->getMethods(\ReflectionMethod::IS_PUBLIC),
+                function (array $dataProviderMethodNames, \ReflectionMethod $method) {
+                    if (false === $method->getDocComment()) {
+                        return $dataProviderMethodNames;
+                    }
+
+                    $docBlock = new DocBlock($method->getDocComment());
+
+                    return array_merge(
+                        $dataProviderMethodNames,
+                        array_map(function (Annotation $annotation) {
```

Comment:
looking from top of this function, how we get here:
foreach -> reduce -> merge -> map

oh my...